examples/src/tls_client_auth/server.rs

```diff
@@ -67,7 +67,7 @@ async fn main() -> Result<(), Box<dyn std::error::Error>> {
 
     let tls = ServerTlsConfig::new()
         .identity(server_identity)
-        .client_ca_root(client_ca_cert);
+        .client_auth(ClientTlsAuth::Mandatory(client_ca_cert));
```

Comment:
Can we update this to be the old behavior and then added `client_auth_mandatory` as additional methods that way we don't break current users?

Reply:
Ok, that is a good suggestion. I would only propose adding `client_auth_optional(bool)` instead, as calling `client_ca_root` causes mandatory authentication by default.

However, this approach brings up the following question: How should it behave when `client_ca_root` is not set and `client_auth_optional(bool)` is called?

One option is to store the `bool` and use it only if `client_ca_root` call is eventually performed, otherwise it would be ignored, but that may be a bit counterintuitive.

The most idiomatic approach, I think, would be to have a single method taking in enum with acceptable configurations (like it is in this PR), but I understand that it would be a breaking change, so I can rework it once we decide on the desired behavior.

Reply:
> One option is to store the bool and use it only if client_ca_root call is eventually performed, otherwise it would be ignored, but that may be a bit counterintuitive.

I think since its a pretty niche use case we can just document it and that should be fine. I don't think its worth it to break users for the small improvement in clarity for a decently uncommon path.

Reply:
Ok, that's a fair point. I reworked it this way.